### PR TITLE
Fix KeyError: 'resources' exception

### PR DESCRIPTION
**Pull request by FrancescoDeSimone**

### CODE REVIEW

File: shrinkwrap.py [hunk 1]
```diff
@@ -460,7 +460,8 @@ def download(args, root):
             else:
                 # This isn't a snap, pull the resource from the appropriate store
                 # use the bundle provided resource revision if available
-                resource_rev = app["resources"].get(resource.name) or resource.revision
+                res = app.get("resources", None)
+                resource_rev = res.get(resource.name) if res else resource.revision
```
Comment (addyess) [inline shrinkwrap.py:464]: I suppose that there was an app without a `resources` key?

how about this instead:

```suggestion
                resource_rev = app.get("resources", {}).get(resource.name)
                resource_rev = resource_rev or resource.revision
```

there's a chance that if either key doesn't exist in this `app["resources"][resource.name]` we want to fall back to `resource.revision`

**Review by addyess — CHANGES_REQUESTED** [review]
Like the idea, small tweak

**Comment by addyess** [comment]
I believe this is addressing https://github.com/charmed-kubernetes/cdk-shrinkwrap/issues/28

**Comment by FrancescoDeSimone** [comment]
> I believe this is addressing #28

Yeah, I forgot to mention that

**Review by addyess — APPROVED** [review]
I'm good with this one.  But i have concerns over the fact that the upstream bundle has applications that are missing resources.  It indicates that building the offline bundle will be a challenge due to the missing resources and trying to replace the appropriately.

I'm working at the moment on a solution to that problem but it may take me a bit to get something worked up

**Pull request merged**